sgxvm/src/precompiles/compliance_bridge.rs

```diff
@@ -45,119 +47,411 @@ impl LinearCostPrecompileWithQuerier for ComplianceBridge {
+                Some(id) => id,
+                None => {
+                    return Err(PrecompileFailure::Revert {
+                        exit_status: ExitRevert::Reverted,
+                        output: encode(&vec![AbiToken::String(
+                            "invalid issuer verification ID".into(),
+                        )]),
+                    });
+                }
+            };
 
-            let user_address = &decoded_params[0];
-            let verification_type = &decoded_params[1];
-            let issuance_timestamp = &decoded_params[2];
-            let expiration_timestamp = &decoded_params[3];
-            let proof_data = &decoded_params[4];
+            let version = match decoded_params[8].clone().into_uint() {
+                Some(ver) => ver.as_u32(),
+                None => {
+                    return Err(PrecompileFailure::Revert {
+                        exit_status: ExitRevert::Reverted,
+                        output: encode(&vec![AbiToken::String(
+                            "invalid version".into(),
+                        )]),
+                    });
+                }
+            };
 
             let encoded_request = coder::encode_add_verification_details_request(
-                user_address.clone().into_address().unwrap(),
+                user_address,
                 caller,
-                verification_type.clone().into_uint().unwrap().as_u32(),
-                issuance_timestamp.clone().into_uint().unwrap().as_u32(),
-                expiration_timestamp.clone().into_uint().unwrap().as_u32(),
-                proof_data.clone().into_bytes().unwrap()
+                origin_chain,
+                verification_type,
+                issuance_timestamp,
+                expiration_timestamp,
+                proof_data,
+                schema,
+                issuer_verification_id,
+                version,
             );
 
             match querier::make_request(querier, encoded_request) {
                 Some(result) => {
-                    let _ = protobuf::parse_from_bytes::<ffi::QueryAddVerificationDetailsResponse>(result.as_slice())
+                    let _ = protobuf::parse_from_bytes::<ffi::QueryAddVerificationDetailsResponse>(
+                        result.as_slice(),
+                    )
+                    .map_err(|_| PrecompileFailure::Revert {
+                        exit_status: ExitRevert::Reverted,
+                        output: encode(&vec![AbiToken::String(
+                            "cannot parse protobuf response".into(),
+                        )]),
+                    })?;
+
+                    Ok((ExitSucceed::Returned, Vec::default()))
+                }
+                None => {
+                    return Err(PrecompileFailure::Revert {
+                        exit_status: ExitRevert::Reverted,
+                        output: encode(&vec![AbiToken::String(
+                            "call to x/compliance failed".into(),
```

Comment:
specify exact error, this string is the same in multiple places, will have a hard time debugging